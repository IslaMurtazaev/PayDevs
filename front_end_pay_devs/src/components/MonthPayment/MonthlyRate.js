import React from "react";
import { Link } from "react-router-dom";
import { history } from "../../index";

const MonthRate = props => {
  return (
    <div>
      <Link to={`${history.location.pathname}/Monthly/${props.monthPayment.id}/workedDay`}>
        <span className="rate">
          {props.rate}
          /per day _____
        </span>
      </Link>
      <button className="removeRate" onClick={() => props.onRemove(props.monthPayment.id)}>X</button>

    </div>
  );
};

export default MonthRate;
