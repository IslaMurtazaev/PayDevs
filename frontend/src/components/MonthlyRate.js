import React from "react";
import { Link } from "react-router-dom";

const MonthlyRate = props => {
  return (
    <div>
      <Link
        to={`/project/${props.projectId}/Monthly/${
          props.monthPayment.id
        }/workedDay`}
        className="workDays"
      >
        <span className="rate">
          {props.monthPayment.rate}
          /per day
        </span>
      </Link>
      <button
        className="removeRate"
        onClick={() => props.onRemove(props.monthPayment.id)}
      >
        X
      </button>
    </div>
  );
};

export default MonthlyRate;
