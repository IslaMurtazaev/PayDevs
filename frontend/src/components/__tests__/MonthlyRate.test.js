import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import MonthlyRate from "../MonthlyRate";

describe("<MonthlyRate />", () => {
  const monthPayment = { id: 1, rate: 150 };

  it("renders rate based on props", () => {
    const component = shallow(<MonthlyRate monthPayment={monthPayment} />);
    expect(component.find(".rate").text()).toBe("150/per day");
  });

  it("renders remove button based on props", () => {
    const onRemoveSpy = jest.fn();
    const component = shallow(<MonthlyRate monthPayment={monthPayment} onRemove={onRemoveSpy} />);
    component.find(".removeRate").simulate("click");
    expect(onRemoveSpy).toBeCalledWith(1);
  });

  it("matches previous snapshot", () => {
    expect(toJson(shallow(<MonthlyRate monthPayment={monthPayment} />))).toMatchSnapshot();
  })
});
