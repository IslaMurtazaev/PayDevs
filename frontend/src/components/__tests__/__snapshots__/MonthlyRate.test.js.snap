// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MonthlyRate /> matches previous snapshot 1`] = `
<div>
  <Link
    className="workDays"
    replace={false}
    to="/project/undefined/Monthly/1/workedDay"
  >
    <span
      className="rate"
    >
      150
      /per day
    </span>
  </Link>
  <button
    className="removeRate"
    onClick={[Function]}
  >
    X
  </button>
</div>
`;
